autopush/base.py

```diff
@@ -45,3 +45,30 @@ def write_error(self, code, **kwargs):
             self.log.failure("Error in handler: %s" % code,
                              client_info=self._client_info)
         self.finish()
+
+    def authenticate_peer_cert(self):
+        """Authenticate the client per the configured peer certs.
+
+        Aborts the request w/ a 401 on failure.
+
+        """
+        cert = self.request.connection.transport.getPeerCertificate()
+        if cert:
+            sha256 = cert.digest('sha256')
+            auth = self.ap_settings.client_certs.get(sha256)
+            if auth is not None:
+                # TLS authenticated
+                self._client_info['tls_auth'] = auth
+                self._client_info['tls_auth_cn'] = cert.get_subject().CN
```

Comment:
At this point a client cert was required and verified. The signature check is 5 lines above, it's incredibly simple: -- if we don't have your verified cert's sha256, you fall out of the if block and are immediately 401'd.